### PR TITLE
Fix TypeError when torch.jit.load is passed a pathlib.Path

### LINKED ISSUE #45824: torch.jit.load does not accept pathlib.Path arguments
## 🐛 Bug

`torch.jit.load` does not accept `pathlib.Path` arguments

## To Reproduce

```python
from pathlib import Path
import torch

model = torch.jit.load(Path("path/to/model.pth"))
```

## Expected behavior

Should be the same as `torch.jit.load("path/to/model.pth")`.

## Current behavior

```
TypeError: import_ir_module(): incompatible function arguments. The following argument types are supported:
    1. (arg0: torch._C.CompilationUnit, arg1: str, arg2: object, arg3: torch._C.ExtraFilesMap) -> torch._C.ScriptModule

Invoked with: <torch._C.CompilationUnit object at 0x7fe633ff4a30>, PosixPath('path/to/model.pth'), None, ExtraFilesMap{}
```

## Environment

PyTorch version: 1.6.0
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.1 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-10ubuntu2) 9.3.0
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: GeForce GTX 1070
Nvidia driver version: 450.51.06
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.2
[pip3] torch==1.6.0
[pip3] torch-model-archiver==0.2.0
[pip3] torchserve==0.2.0
[conda] numpy                     1.19.2                   pypi_0    pypi
[conda] torch                     1.6.0                    pypi_0    pypi
[conda] torch-model-archiver      0.2.0                    pypi_0    pypi
[conda] torchserve                0.2.0                    pypi_0    pypi

cc @gmagogsfm

### PR DESCRIPTION
**Pull request by jonathan-conder-sm**
Fixes #45824

### CODE REVIEW

**Review by gmagogsfm — CHANGES_REQUESTED** [review]
Could you add a test case for this? Thank you

**Comment by jonathan-conder-sm** [comment]
Turns out `torch.jit.save` didn't work either. I fixed that & added a test for both methods

**Review by gmagogsfm — APPROVED** [review]
Looks good, please fix python lint. Let me know when it is ready, and I will commit for you.

File: torch/jit/_script.py [hunk 1]
```diff
@@ -478,13 +478,13 @@ def code_with_constants(self):
             r = self.forward.code_with_constants
             return (r[0], ConstMap(r[1]))
 
-        def save(self, f, _extra_files={}):
+        def save(self, f, **kwargs):
```
Comment (gmagogsfm) [inline torch/jit/_script.py:481]: OOC, why is this needed?

Comment (jonathan-conder-sm) [reply torch/jit/_script.py:481]: The lint error was because I was passing a mutable object (i.e. `{}`) as a default argument. Alternative solution would be to have `None` as the default with
```python
if _extra_files is None:
    _extra_files = {}
```
inside the function. But before I made any changes the function was already using kwargs so I figured I'd just switch back to that.

**Comment by jonathan-conder-sm** [comment]
Think it's ready now. Thanks!
